kategory-docs/docs/docs/optics/fold/README.md

```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
 
 ## Fold
 
-TODO
\ No newline at end of file
+`Fold` is an optic that allows to focus into structure and get multiple results.
+It is a generalisation of something `Foldable`.
+
+Creating a `Fold` can be done by manually defining `foldMap`.
+
+```kotlin:ank
+import kategory.*
+import kategory.optics.*
+
+fun <T> nullableFold(): Fold<T?, T> = object : Fold<T?, T> {
+    override fun <R> foldMap(M: Monoid<R>, s: T?, f: (T) -> R): R =
+        s?.let(f) ?: M.empty()
+}
+```
+
+Or you can get a `Fold` from any existing `Foldable`.
+
+```kotlin:ank:silent
+val nonEmptyIntFold: Fold<NonEmptyListKind<Int>, Int> = Fold.fromFoldable(NonEmptyList.foldable())
+```
+
+`Fold` has a similar API as `Foldable` but because it's defined in terms of `foldMap` there are no associative fold functions available.
+
+```kotlin:ank
+nullableFold<Int>().isEmpty(null)
+```
+```kotlin:ank
+nonEmptyIntFold.combineAll(NonEmptyList.of(1, 2, 3))
+```
+```kotlin:ank
+nullableFold<Int>().headOption(null)
+```
+```kotlin:ank
+nonEmptyIntFold.headOption(NonEmptyList.of(1, 2, 3, 4))
+```
+
+## Composition
+
+Composing `Fold` can be used for accessing nested structures to be able to see foci.
```

Comment:
`to be able to see foci` I don't quite understand what foci means

Reply:
What I mean to say here is that if for example you have a Fold for `List<List<_>>` that can "focus" into `List<_>` and you compose it with a Fold for `List<Int>` with a focus in `Int` you get back a Fold with many focuses (foci). But now that I write it like this I am unsure if "foci" makes sense here.

Reply:
`Composing `Fold` allows the user to create an aggregated foci (?) that focuses on nested structures`. Something in these terms maybe.